table/tables/partition.go

```diff
@@ -1025,10 +1024,7 @@ func (t *partitionedTable) locateRangeColumnPartition(ctx sessionctx.Context, pi
 	if err != nil {
 		return 0, errors.Trace(err)
 	}
-	if isNull {
-		idx = 0
-	}
-	if idx < 0 || idx >= len(partitionExprs) {
+	if idx >= len(partitionExprs) {
```

Comment:
If the value from the row is NULL (or really if the partitioning expression evaluates to NULL) it is already compared during the search and idx will already become 0.
A partition definition value (`partitionExprs`) can never be NULL, so the generated expression (comparison between partition definition value and column (or partition expression)) can only be NULL if the column is NULL (or the partition expression evaluates to NULL.

I wonder if we should also propagate if there is an error during the Search? Like creating a `hadError bool` variable and setting it to true if `err != nil` and use that at line 1025/1024 instead, since the current `err` is only from the last call during the binary `sort.Search`. (Actually the same affects the isNull value anyway...)

Reply:
What if `idx < 0`?

Reply:
[sort.Search()](https://pkg.go.dev/sort#Search) return values in the range of [0,n), so it cannot happen.

Reply:
oh, look like `idx` won't be -1.

Reply:
Got it, thanks!